File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/mediation/AdMobBannerFragment.java
package tv.teads.teadssdkdemo.format.mediation;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.Toast;

import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdSize;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;

import org.greenrobot.eventbus.Subscribe;

import tv.teads.adapter.admob.TeadsAdNetworkExtras;
import tv.teads.adapter.admob.TeadsAdapter;
import tv.teads.teadssdkdemo.R;
import tv.teads.teadssdkdemo.utils.BaseFragment;
import tv.teads.teadssdkdemo.utils.ReloadEvent;

/**
 * Display inRead as Banner within a ScrollView using AdMob Mediation.
 */
public class AdMobBannerFragment extends BaseFragment {

    // This is is should be replaced by your own AdMob application and ad block ids
    public static final String ADMOB_TEADS_APP_ID = "ca-app-pub-3570580224725271~3869556230";
    public static final String ADMOB_TEADS_BANNER_ID = "ca-app-pub-3570580224725271/1481793511";

    private FrameLayout mBannerAdFrame;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_admob_banner, container, false);
        mBannerAdFrame = rootView.findViewById(R.id.banner_ad_frame);
        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        // 1. Init AdMob (could be done in your Application class)
        MobileAds.initialize(getContext(), ADMOB_TEADS_APP_ID);

        // 2.Create AdMob viewa nd add it to hierarchy
        AdView adView = new AdView(view.getContext());
        adView.setAdUnitId(ADMOB_TEADS_BANNER_ID);
        adView.setAdSize(AdSize.MEDIUM_RECTANGLE);
        mBannerAdFrame.addView(adView);

        // 3. Attach listener (will include Teads callbacks)
        adView.setAdListener(new AdListener() {
            @Override
            public void onAdLoaded() {
                // maybe track it on GA?
            }

            @Override
            public void onAdFailedToLoad(int errorCode) {
                Toast.makeText(getContext(), "Ad loading failed: onAdFailedToLoad(" + errorCode + ")", Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onAdOpened() {
                // Nothing to do
            }

            @Override
            public void onAdLeftApplication() {
                // Nothing to do
            }

            @Override

            public void onAdClosed() {
                // Nothing to do
            }
        });

        // 4. Load a new ad (this will call AdMob that will and Teads afterward)
        TeadsAdNetworkExtras extras = new TeadsAdNetworkExtras.Builder()
                                              // Needed by european regulation
                                              // See https://mobile.teads.tv/sdk/documentation/android/gdpr-consent
                                              .userConsent("1", "0001")
                                              // The article url if you are a news publisher
                                              .pageUrl("https://page.com/article1/")
                                              .build();
        AdRequest adRequest = new AdRequest.Builder()
                                      .addCustomEventExtrasBundle(TeadsAdapter.class, extras.getExtras())
                                      .build();

        adView.loadAd(adRequest);
    }


    @Subscribe
    @SuppressWarnings("unused")
    public void onReloadEvent(ReloadEvent event) {
        // not used
    }

}

File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/mediation/AdMobBannerFragment.kt
package tv.teads.teadssdkdemo.format.mediation

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.google.android.gms.ads.*
import kotlinx.android.synthetic.main.fragment_admob_banner.*
import org.greenrobot.eventbus.Subscribe
import tv.teads.adapter.admob.TeadsAdapter
import tv.teads.helper.TeadsBannerAdapterListener
import tv.teads.helper.TeadsHelper
import tv.teads.sdk.android.AdSettings
import tv.teads.teadssdkdemo.R
import tv.teads.teadssdkdemo.utils.BaseFragment
import tv.teads.teadssdkdemo.utils.ReloadEvent
import kotlin.math.roundToInt

/**
 * Display inRead as Banner within a ScrollView using AdMob Mediation.
 */
class AdMobBannerFragment : BaseFragment() {
    private lateinit var mListener: TeadsBannerAdapterListener

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_admob_banner, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        // 1. Init AdMob (could be done in your Application class)
        MobileAds.initialize(context, ADMOB_TEADS_APP_ID)
        TeadsHelper.initialize()

        // 2. Create AdMob view and add it to hierarchy
        val adView = AdView(view.context)
        adView.adUnitId = ADMOB_TEADS_BANNER_ID
        adView.adSize = AdSize.MEDIUM_RECTANGLE
        bannerAdFrame.addView(adView)

        // 3. Attach listener (will include Teads events)
        adView.adListener = object : AdListener() {
            override fun onAdLoaded() {
                // maybe track it on GA?
            }

            override fun onAdFailedToLoad(errorCode: Int) {
                Toast.makeText(context, "Ad loading failed: onAdFailedToLoad($errorCode)", Toast.LENGTH_SHORT).show()
            }

            override fun onAdOpened() {
                // Nothing to do for Teads
            }

            override fun onAdLeftApplication() {
                // Nothing to do for Teads
            }

            override fun onAdClosed() {
                // Nothing to do for Teads
            }
        }

        mListener = object : TeadsBannerAdapterListener {
            override fun onRatioUpdated(adRatio: Float) {
                val params: ViewGroup.LayoutParams = adView.layoutParams

                // Here the width is MATCH_PARENT
                params.height = (params.width / adRatio).roundToInt()

                adView.layoutParams = params
            }

        }

        // 4. Attach listener to the helper and save the key
        val key = TeadsHelper.attachListener(mListener)

        // 5. Load a new ad (this will call AdMob and Teads afterward)
        val extras = AdSettings.Builder()
                // Needed by european regulation
                // See https://mobile.teads.tv/sdk/documentation/android/gdpr-consent
                .userConsent("1", "0001")
                // The article url if you are a news publisher to increase your earnings
                .pageUrl("https://page.com/article1/")
                .addAdapterListener(key)
                .build()
        val adRequest = AdRequest.Builder()
                .addCustomEventExtrasBundle(TeadsAdapter::class.java, extras.toBundle())
                .build()

        adView.loadAd(adRequest)
    }


    @Suppress("UNUSED_PARAMETER")
    @Subscribe
    fun onReloadEvent(event: ReloadEvent) {
        // not used
    }

    companion object {
        // FIXME This ids should be replaced by your own AdMob application and ad block ids
        val ADMOB_TEADS_APP_ID = "ca-app-pub-3068786746829754~3613028870"
        val ADMOB_TEADS_BANNER_ID = "ca-app-pub-3068786746829754/3486435166"
    }

}
